Cookie based authentication

spwoodcock opened the pull request:
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Issue

As discussed in https://github.com/hotosm/fmtm/discussions/1083

## Describe this PR

- We had authentication in place to return the OSM user details.
- However the **authentication** details did not persist, so implementing **authorisation** would require reacquiring **authentication** every time.
- This PR adds support for storing the auth token (OSM JWT token) in a secure cookie.
- Cookie details: samesite:lax (for CSRF), httpOnly (for XSS), secure (for MIM).

## Screenshots

![image](https://github.com/hotosm/fmtm/assets/78538841/10fd8763-81a3-48ec-81db-d0945d812b80)

## Implications

- For testing the api using auth, we no longer have to pass the access_token manually.
- If you login once via the localhost frontend, then a cookie will be saved in your browser.
- This cookie will be automatically extracted when testing endpoints on the docs page: http://api.fmtm.localhost:7050/docs making development a lot smoother.
fyi @nrjadkry @Sujanadh 

## Checklist before requesting a review

- 📖 Read the FMTM Contributing Guide: <https://github.com/hotosm/fmtm/blob/main/CONTRIBUTING.md>
- 📖 Read the FMTM Code of Conduct: <https://github.com/hotosm/fmtm/blob/main/CODE_OF_CONDUCT.md>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

## [optional] What gif best describes this PR or how it makes you feel?

![giphy](https://github.com/hotosm/fmtm/assets/78538841/5a4dfd01-1729-444d-b554-e51e2548aed5)

varun2948 commented:
Wow @spwoodcock loved the new PR template :smiley: and with the GIF addition too. Nice :100:

varun2948 commented:
Can we add how the commit message should be too ? any docs or any url to follow commit message pattern cause at naxa we do `Feat: feature description`

spwoodcock commented:
We have some basic info about conventional commit messages here: https://docs.hotosm.org/dev-guide/version-control/#conventional-commits

I guess it depends on if we merge commits are they are, or squash them within a PR.
Until now, we have squashed commits in a PR into a single commit.

So when merging we should be sure to update the squashed commit message to use conventional commits too: `feat: added some new feature in the PR`
